src/daemon/ubuntu_image_host.cpp

```diff
@@ -229,7 +230,11 @@ void mp::UbuntuVMImageHost::fetch_manifests()
             manifests.emplace_back(
                 std::make_pair(remote.first, download_manifest(QString::fromStdString(remote.second), url_downloader)));
         }
-        catch (mp::DownloadException& e)
+        catch (mp::EmptyManifestException& /* e */)
+        {
+            on_manifest_empty(fmt::format("Did not find any supported products in \"{}\"", remote.first));
+        }
+        catch (std::runtime_error& e)
```

Comment:
Hi, shouldn't this be DownloadException?

Reply:
I saw this, but thought the intent was to catch all other errors instead with the change.  Let's see what @Saviq says :grin:

Reply:
No, the intent was to not bail on any of the errors (one broken remote shouldn't cause all the other ones).

The EmptyManifestException was needed to not warn about that particular case.

Reply:
Uh, I must be looking at this wrong, but isn't a `DownloadException` a `runtime_error`? So `DownloadException`s will still be treated as before, but now so will any other `runtime_error`s? Or was the error you wanted to continue on not a `DownloadException` in the first place? In that case, it should be another specific error, so that we're not catching undesired stuff here, right?

Reply:
Sure, I agree catching `std::runtime_error` is dangerously broad. It's just that's what `SimpleStreamsManifest::fromJson` is throwing. I'll fix.

bors cancel